File: src/visualization.cpp
#include "visualization.h"
#include <iostream>
#include <vector>
#include <matplotlibcpp.h>

namespace plt = matplotlibcpp;

void plotFields(const std::vector<double>& ey, const std::vector<double>& bz, double time) {
    plt::figure_size(800, 600);
    
    plt::subplot(2, 1, 1);
    plt::title("Electric Field (Ey) at time " + std::to_string(time));
    plt::xlabel("Position");
    plt::ylabel("Ey");
    plt::plot(ey);
    
    plt::subplot(2, 1, 2);
    plt::title("Magnetic Field (Bz) at time " + std::to_string(time));
    plt::xlabel("Position");
    plt::ylabel("Bz");
    plt::plot(bz);
    
    plt::show();
}

void visualize(const std::vector<double>& ey, const std::vector<double>& bz, double time) {
    plotFields(ey, bz, time);
}

File: src/electric_wave_py/visualization.py
class Visualization:
    def __init__(self):
        import matplotlib.pyplot as plt
        self.plt = plt

    def plot_fields(self, ey, bz, time_step):
        self.plt.figure(figsize=(12, 6))

        self.plt.subplot(1, 2, 1)
        self.plt.title(f'Electric Field (ey) at t={time_step}')
        self.plt.plot(ey, color='blue')
        self.plt.xlabel('Position')
        self.plt.ylabel('Electric Field (ey)')
        self.plt.grid()

        self.plt.subplot(1, 2, 2)
        self.plt.title(f'Magnetic Field (bz) at t={time_step}')
        self.plt.plot(bz, color='red')
        self.plt.xlabel('Position')
        self.plt.ylabel('Magnetic Field (bz)')
        self.plt.grid()

        self.plt.tight_layout()
        self.plt.show()